service/block/fraud.go

```diff
@@ -0,0 +1,7 @@
+package block
+
+// badEncodingCheck handles checking the given ExtendedBlock against
+// the given Raw block and generating fraud proofs if necessary. // TODO @renaynay: this method is stubbed out for now.
+func (s *Service) badEncodingCheck(extendedBlock *ExtendedBlock, raw *Raw) error {
```

Comment:
I think this method should be called `validateEncoding` and it should take the extended block and the header (extended if at hand?) as input. There should be a BadEncodingError struct that contains all relevant info to create a bad encoding fraud-proof (for now that struct could also just be empty).

Don't worry about it for now. We should spec that out first anyways.

Reply:
Okay yeah let's try and spec it out so I can at least implement the structures.

Reply:
Btw which header? The header from the raw block or the generated `ExtendedHeader` ? bc the generated `ExtendedHeader` is the one we produce on celestia node side after extending block, so then how would you compare to the data root that's sent over by the core node?

Reply:
>  bc the generated ExtendedHeader is the one we produce on celestia node side after extending block, so then how would you compare to the data root that's sent over by the core node?

The block you get from tendermint contains a header, which contains a data root. That root must match the DAHeader's hash which you compute locally.

Reply:
that's what I thought